### ISSUE TITLE
permute for 3 dim tensor

### ISSUE THREAD

**Issue by yli1994** [NVIDIA-AI-IOT/torch2trt#349]
Hi jaybdub,

If I got a tensor which has x.shape = [1,1024,22], and I want to permute the dims with x = x.permute(2,0,1) but it went wrong(AssertionError).What should I do to get the same result and the process can be transferred to tensorrt?

**Comment by jaybdub** [comment]
Hi yli1994,

Thanks for reaching out!

torch2trt assumes the first dimension corresponds to the batch size, which is determined at run-time.  Unfortunately, it's not possible with torch2trt to permute the batch dimension.  For example, something like ``x = x.permute(0, 2, 1)`` would work because it does not modify the batch dimension location.

Are you able to share more information about the model you need to perform this for?  Perhaps there is a way around this limitation.

Best,
John

**Comment by yurui777** [comment]
Like in model(OCR-CRNN)，before the rnn, we need change the shape of (b,c,w) to (w,b,c),so we use permute(2,0,1).Then, assert error!

**Comment by jaybdub** [comment]
Hmm, I see.  I wonder, have you explored the ``batch_first`` option in PyTorch?

https://pytorch.org/docs/master/generated/torch.nn.RNN.html

Perhaps it's possible to accomplish something there.  

Alternatively, if the RNN is part of a larger submodule, say

```python
class RnnSubmodule():
    def forward(self, x):
        # reshape to seq first (b,c,w) -> (w,b,c)
        # run RNN
        # reshape back to batch first
```

Then you could potentially write a converter for the larger submodule using a method like here

https://github.com/NVIDIA-AI-IOT/torch2trt/pull/175

Please forgive, this pull request is experimental, but I'd be curious to hear if this helps your use case.

Also general apologies, I'm not too experienced with RNN practices.

Best,
John

**Comment by yli1994** [comment]
Hi @jaybdub ,
Thank you for your answer, I will try what you said!
